fecgov/fec-cms
# Rename ReportingExamplePage to Example Page, add features

## Summary
Rename ReportingExamplePage to Example Page, add features
        - Resolves #3149

- Rename ReportingExample page to ExamplePage
- makemigrations/migrate
- Add new features to ExamplePage 
- makemigrations
- Restore local database to latest dump
- Combine the operations in the last two migrations above by manually editing the first to include operatons from the second * and then delete the second file .(rename page first, then add new features.) This way I only push one new migration file up.
- migrate again
***Note:** If I had renamed the model and added new features in the same makemigrations action, Django would have considered the ExamplePage as a whole new model and deleted the ReportingExamplePage model , making existing pages based on that would not work, as I found in testing. If the model is exactly the same but has a new name, Django asks if you intended to just rename it. Then you can add your new features to the model and migrate again without breaking pages created  under the old name.

- [ ] Followup issue: Its probably a good time to squash migrations. See https://github.com/fecgov/fec-cms/issues/3094 and [Django Docs](https://docs.djangoproject.com/en/2.2/topics/migrations/#squashing-migrations): 

## Impacted areas of the application
new file:   home/migrations/0106_auto_20191101_1759.py
modified:   home/models.py
modified:   home/templates/home/reporting_example_page.html


## How to test
**Content team:** This can be tested on feature space https://fec-feature-cms.app.cloud.gov/admin
Don't forget to test existing pages that were created originally with ReportingExampePage before it was renamed:
Go to `Add Child Page` > on the far right column of the page types listing you can click on `Pages using Example page` and spot check that these pages match what is on production

**Developers:** 
1) Checkout `feature/3149-ExamplePage-template-wagtail`
2) `./manage.py migrate`
3) Go to Wagtail admin: http://localhost:8000/admin
4) Create and preview/publish a page of type ExamplePage
5) Test the image, external-link, internal-link blocks
6) Embed a video using the Paragraph blocks RTE embed media option
7) It is also important to spot-check existing pages that were created originally with ReportingExamplePage template, before it was renamed.
Go to `Add Child Page` > on the far right column of the page types listing you can click on `Pages using Example page` and spot-check that these pages match what is on production
